pytorch_lightning/accelerators/accelerator.py

```diff
@@ -213,6 +213,7 @@ def __getstate__(self):
         }
 
     def __setstate__(self, d):
+        assert isinstance(d, dict), f"{type(d)} passed instead of dict."
```

Comment:
can you explain why this is needed? Lightning never calls this internal method, it's for pickling yes?